plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordCursor.java

```diff
@@ -276,8 +276,8 @@ public void close()
                     // statement already closed or cancel is not supported
                 }
             }
-            if (connection != null) {
-                jdbcClient.abortReadConnection(connection);
+            if (connection != null && resultSet != null) {
```

Comment:
Do we have a race condition here? What if another thread is running `advanceNextPosition()`? Then it could be a case that we almost have result set and we won't abort the connection.

Reply:
Well, I'd need to inspect the code more closely, but if there is a race condition there, then it was also there before. Note that we're using a local variable initialized from the `resultSet` field in the `try` initializer, so if some thread creates the result set after we start closing, it won't get cleaned up.

Reply:
Previously we were calling `abortReadConnection` even when `resultSet = null`, can we restore this?

Reply:
I don't think it makes sense - if there's no `ResultSet` then there's no buffered data and nothing to drain, and so calling `abort` is not necessary